src/javatests/com/google/cloud/bigquery/dwhassessment/hooks/logger/EventRecordConstructorTest.java

```diff
@@ -223,6 +223,11 @@ public void postExecHook_recordsYarnApplicationDataWhenPossible() {
     report.setStartTime(100000L);
     report.setFinishTime(200000L);
     report.setFinalApplicationStatus(FinalApplicationStatus.UNDEFINED);
+    String expectedJson =
+        "[{\"YarnApplicationId\":\"application_1685098059769_1951\",\"HiveHostName\":\"test_host\",\"Queue\":\"test_queue\",\"YarnProcess\":\"10000.6\",\"YarnApplicationType\":\"TEZ\",\"YarnApplicationState\":\"RUNNING\",\"YarnDiagnostics\":\"test_diagnostics\",\"YarnCurrentApplicationAttemptId\":\"appattempt_1685098059769_1951_000001\",\"YarnUser\":\"test_user\",\"YarnStartTime\":\"100000\",\"YarnFinishTime\":\"200000\",\"YarnFinalApplicationStatus\":\"UNDEFINED\",\"YarnReportNumUsedContainers\":\"3\",\"YarnReportNumReservedContainers\":\"4\",\"YarnReportMemorySeconds\":\"100\",\"YarnReportVcoreSeconds\":\"400\",\"YarnReportQueueUsagePercentage\":\"90.5\",\"YarnReportClusterUsagePercentage\":\"64.5\",\"YarnReportPreemptedMemorySeconds\":\"200\",\"YarnReportPreemptedVcoreSeconds\":\"300\",\"YarnReportUsedResources\":\"<memory:800,"
```

Comment:
Can we please have tests with 0 and 2 elements in the JSON array?

Reply:
added test for 2 elements, will add 0 elements test in separate pull request

Reply:
added test for 0 element case in this pull request